06-job-control.md

```diff
@@ -4,8 +4,169 @@ title: Extra Unix Shell Material
 subtitle: Job control
 minutes: 5
 ---
+
 > ## Learning Objectives {.objectives}
 >
 > * FIX ME
 
-FIX ME
\ No newline at end of file
+Our next topic is how to control programs *once they're running*. This
+is called [job control](glossary.html#job-control), and while it's less
+important today than it was back in the Dark Ages, it is coming back
+into its own as more people begin to leverage the power of computer
+networks.
+
+When we talk about controlling programs, what we really mean is
+controlling *processes*. As we said earlier, a process is just a program
+that's in memory and executing. Some of the processes on your computer
+are yours: they're running programs you explicitly asked for, like your
+web browser. Many others belong to the operating system that manages
+your computer for you, or, if you're on a shared machine, to other
+users. You can use the `ps` command to list them, just as you use `ls`
+to list files and directories:
+
+~~~{.input}
+$ ps
+~~~
+~~~{.output}
+PID    PPID   PGID   TTY   UID     STIME   COMMAND
+2152      1   2152   con  1000  13:19:07   /usr/bin/bash
+2276   2152   2276   con  1000  14:53:48   /usr/bin/ps
+~~~
+
+Every process has a unique process id (PID). Remember, this is a
+property of the process, not of the program that process is executing:
+if you are running three instances of your browser at once, each will
+have its own process ID.
+
```

Comment:
Is it worth noting that PIDs fall into a certain range and can "overflow" i.e. the same PID can refer to different processes over the uptime of the OS.